pkg/services/signingkeys/signingkeysimpl/service.go

```diff
@@ -0,0 +1,90 @@
+package signingkeysimpl
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+
+	"github.com/go-jose/go-jose/v3"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/signingkeys"
+)
+
+const (
+	serverPrivateKeyID = "default"
+)
+
+var _ signingkeys.Service = new(Service)
+
+func ProvideEmbeddedSigningKeysService() *Service {
+	s := &Service{
+		log: log.New("auth.key_service"),
+	}
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		s.log.Error("Error generating private key", "err", err)
```

Comment:
I think we should error out if the server is not able to generate its private key, wdyt?

Reply:
Done